[MM-12340] update release process (#182)
mattermost/mattermost-developer-documentation: site/content/internal/release-process.md

# Release Cutting Process

**Note: To cut a build you need access to matterbuild. Please ask Christopher/Carlos for access if you don't have it.**

> Developers and PMs decides when to create the release branch. The branch can be created from master or from a existing release branch, it depends for which type of release we are planning. If is a Quality Release the new branch should be branch out from a previous release, otherwise the branch should happen from the master branch.

Reviewer: ```suggestion
Developers and PMs decide when to create the release branch. The branch can be created from master or from an existing release branch, it depends which type of release we are planning. If it is a Quality Release, the new branch should be branched off the previous release, otherwise the branch should be off the master branch.
```

Minor grammar suggesstions

> All commits goes to master branch and then cherry-pick to the appropriate branch. Commits to master follow the PR process, once this is approved and merge the developer should cherry-pick that commit to the appropriate branch. No PR are needed.

Reviewer: ```suggestion
All commits go to master branch and then are cherry-picked to the appropriate branch. Commits to master follow the PR process, once approved and merged the developer should cherry-pick that commit to the appropriate branch. No PR is needed.
```
